crypto/common/fdoKeyExchange.c

```diff
@@ -252,26 +256,26 @@ static int32_t prep_keymat(uint8_t *keymat, size_t keymat_size, const int index,
 		goto err;
 	}
 
-	keymat[ofs++] = idx0_val;
+	kdf_input[ofs++] = idx0_val;
 	// Fill in the kdflabel
-	if (strncpy_s((char *)&keymat[ofs], keymat_size - ofs,
+	if (strncpy_s((char *)&kdf_input[ofs], kdf_input_len - ofs,
 		kex_ctx->kdf_label,
 		strnlen_s(kex_ctx->kdf_label, FDO_MAX_STR_SIZE))) {
 		LOG(LOG_ERROR, "Failed to fill kdf label in key Material\n");
 		goto err;
 	}
 	ofs += strnlen_s(kex_ctx->kdf_label, FDO_MAX_STR_SIZE);
-	keymat[ofs++] = 0x00;
+	kdf_input[ofs++] = 0x00;	// separation indicator
```

Comment:
For uniformity, add the comment line above the actual statement.

// separation indicator
kdf_input[ofs++] = 0x00;